javagen/src/main/java/com/azure/autorest/util/TemplateUtil.java

```diff
@@ -124,24 +133,25 @@ public static void writeClientMethodsAndHelpers(JavaClass classBlock, List<Clien
     public static void writeLongRunningOperationTypeReference(JavaBlock javaBlock, ClientMethod clientMethod) {
         // see writeTypeReferenceStaticClass
 
-        if (clientMethod.getMethodPollingDetails().getIntermediateType() instanceof ClassType && clientMethod.getMethodPollingDetails().getFinalType() instanceof ClassType) {
-            // use static inner class
-            javaBlock.line("new TypeReference%s(), new TypeReference%s());",
-                    clientMethod.getMethodPollingDetails().getIntermediateType(),
-                    clientMethod.getMethodPollingDetails().getFinalType());
-        } else {
-            javaBlock.line("new TypeReference<%s>() {\n// empty\n}, new TypeReference<%s>() {\n// empty\n});",
-                    clientMethod.getMethodPollingDetails().getIntermediateType(),
-                    clientMethod.getMethodPollingDetails().getFinalType());
-        }
+        MethodPollingDetails details = clientMethod.getMethodPollingDetails();
+        javaBlock.line("%s, %s);", getTypeReferenceCreation(details.getIntermediateType()),
```

Comment:
`%s, %s);` here seems not right. (this line might not be used anywhere in test)

Reply:
This is following what it was using before, being either

`javaBlock.line("new TypeReference%s(), new TypeReference%s());`

or

`javaBlock.line("new TypeReference<%s>() {\n// empty\n}, new TypeReference<%s>() {\n// empty\n});"`

but I agree that this is definitely a spot where a lot of bugs could crop up as it's making an assumption that this is always going to be inserting the last two parameters in an API call.

Reply:
OK, this might be old problem of mine. I will try improve it (make the "writeLongRunningOperationTypeReference" return a good expression)